aiida/engine/processes/calcjobs/calcjob.py

```diff
@@ -527,10 +527,6 @@ def parse_scheduler_output(self, retrieved: orm.Node) -> Optional[ExitCode]:
             scheduler_stdout = None
             self.logger.warning(f'could not parse scheduler output: the `{filename_stdout}` file is missing')
 
-        # Only attempt to call the scheduler parser if all three resources of information are available
-        if any(entry is None for entry in [detailed_job_info, scheduler_stderr, scheduler_stdout]):
```

Comment:
I think we could replace this with

```python
if all(entry is None for entry in [detailed_job_info, scheduler_stderr, scheduler_stdout]):
```

Reply:
Furthermore, while I understand that `detailed_job_info` can be `None`, I wonder whether it is ever legitimate for `stdout` or `stderr` to be `None`?

`node.get_option()` will return `None`, if the option is missing, but is that not an indication that something is wrong?
Perhaps a warning should be logged in that case (not necessary for this PR, though).

Reply:
A warning is being logged if either is `None` on line 522 and 528 respectively. In the spirit of this PR, I think we should keep it at a warning and continue calling `parse_output` and leave it up to the plugins to try and extract as much information as possible.

Reply:
But I think you make a good point about `node.get_option` returning `None` if the option was never set, because it is being passed straight to `retrieved.get_object_content` and this should result in a `TypeError` being thrown from `Repository.get_file` which doesn't accept `None`. This should also be caught.